lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
-      'in NxOveflowTooltip', function() {
+  it('renders the children within the NxDropdownMenu in the specified order', async function() {
     const children = [
-      <a id="link1" key="1">Link1</a>,
-      <a id="link2" className="nx-dropdown-link" key="2">Link2</a>,
-      <button id="link3" className="nx-dropdown-button" key="3">Link3</button>,
-      <button id="link4" className="nx-dropdown-right-button" key="4">Link4</button>
+      <a data-testid="menu-child" id="link1" key="1">Link1</a>,
+      <a data-testid="menu-child" id="link2" className="nx-dropdown-link" key="2">Link2</a>,
+      <button data-testid="menu-child" id="link3" className="nx-dropdown-button" key="3">Link3</button>,
+      <button data-testid="menu-child" id="link4" className="nx-dropdown-right-button" key="4">Link4</button>
     ];
-    const component = getMountedComponent({ children, isOpen: true }),
-        menu = component.find('div.nx-dropdown-menu');
-
-    expect(menu.childAt(0)).toMatchElement(<a id="link1" key="1">Link1</a>);
 
-    expect(menu.childAt(1)).toMatchSelector(NxOverflowTooltip);
-    expect(menu.childAt(1).find('a')).toMatchElement(<a id="link2" key="2" className="nx-dropdown-link">Link2</a>);
+    const { queryAllByTestId } = quickRender({ children, isOpen: true });
 
-    expect(menu.childAt(2)).toMatchSelector(NxOverflowTooltip);
-    expect(menu.childAt(2).find('button')).toMatchElement(
-      <button id="link3" key="3" className="nx-dropdown-button">Link3</button>
-    );
+    const menuChildren = queryAllByTestId('menu-child');
 
-    expect(menu.childAt(3)).toMatchElement(
-      <button id="link4" key="4" className="nx-dropdown-right-button">Link4</button>
-    );
+    expect(menuChildren[0] as HTMLElement).toHaveTextContent('Link1');
+    expect(menuChildren[1] as HTMLElement).toHaveTextContent('Link2');
+    expect(menuChildren[2] as HTMLElement).toHaveTextContent('Link3');
+    expect(menuChildren[3] as HTMLElement).toHaveTextContent('Link4');
   });
 
-  it('calls onToggleCollapse if a click happens anywhere when the dropdown is already open', function() {
-    const onToggleCollapse = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse, isOpen: true }, { attachTo: container });
+  it('calls onToggleCollapse if a click happens anywhere when the dropdown is already open', async function() {
+    const user = userEvent.setup();
+    const onToggleCollapse = jest.fn();
+    const { rerender } = quickRender({ onToggleCollapse, isOpen: true });
 
     expect(onToggleCollapse).not.toHaveBeenCalled();
 
-    act(() => {
-      document.dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    await user.click(document.body);
+    rerender(<NxDropdown {...minimalProps} onToggleCollapse={onToggleCollapse} isOpen={true} />);
+
     expect(onToggleCollapse).toHaveBeenCalled();
```

Comment:
Might be a good idea to check the number of calls here too, since the toggle code is known to be kind of touchy.
```suggestion
    expect(onToggleCollapse).toHaveBeenCalledTimes(1);
```